devicetypes/fibargroup/fibaro-wall-plug-us-zw5.src/fibaro-wall-plug-us-zw5.groovy

```diff
@@ -0,0 +1,496 @@
+
+	def cmds = []
+	log.warn "Executing updated"
+	if (!childDevices) {
+		createChildDevices()
+	}
+	state.lastUpdated = now()
+	syncStart()
+}
+
+def configure() {
+	def cmds = []
+	cmds << zwave.multiChannelAssociationV2.multiChannelAssociationGet(groupingIdentifier:1)
+	cmds << zwave.basicV1.basicSet(value: 0)
+	encapSequence(cmds,1000)
+}
+
+def ping() {
+	refresh()
+}
+
+private syncStart() {
+	boolean syncNeeded = false
+	Integer settingValue = null
+	parameterMap().each {
+		if(settings."$it.key" != null) {
+			settingValue = settings."$it.key" as Integer
+			if (state."$it.key" == null) { state."$it.key" = [value: null, state: "synced"] }
+			if (state."$it.key".value != settingValue || state."$it.key".state != "synced" ) {
+				state."$it.key".value = settingValue
+				state."$it.key".state = "notSynced"
+				syncNeeded = true
+			}
+		}
+	}
+	if ( syncNeeded ) {
+		logging("sync needed.", "info")
+		syncNext()
+	}
+}
+
+private syncNext() {
+	logging("Executing syncNext()","info")
+	def cmds = []
+	for ( param in parameterMap() ) {
+		if ( state."$param.key"?.value != null && state."$param.key"?.state in ["notSynced","inProgress"] ) {
+			multiStatusEvent("Sync in progress. (param: ${param.num})", true)
+			state."$param.key"?.state = "inProgress"
+
+			cmds << response(encap(zwave.configurationV2.configurationSet(scaledConfigurationValue: state."$param.key".value, parameterNumber: param.num, size: param.size)))
+			cmds << response(encap(zwave.configurationV2.configurationGet(parameterNumber: param.num)))
+			break
+		}
+	}
+	if (cmds) {
+		runIn(10, "syncCheck")
+		sendHubCommand(cmds,1000)
+	} else {
+		runIn(1, "syncCheck")
+	}
+}
+
+private syncCheck() {
+	logging("Executing syncCheck()","info")
+	def failed = []
+	def incorrect = []
+	def notSynced = []
+	parameterMap().each {
+		if (state."$it.key"?.state == "incorrect" ) {
+			incorrect << it
+		} else if ( state."$it.key"?.state == "failed" ) {
+			failed << it
+		} else if ( state."$it.key"?.state in ["inProgress","notSynced"] ) {
+			notSynced << it
+		}
+	}
+
+	if (failed) {
+		multiStatusEvent("Sync failed! Verify parameter: ${failed[0].num}", true, true)
+	} else if (incorrect) {
+		multiStatusEvent("Sync mismatch! Verify parameter: ${incorrect[0].num}", true, true)
+	} else if (notSynced) {
+		multiStatusEvent("Sync incomplete! Open settings and tap Done to try again.", true, true)
+	} else {
+		if (device.currentValue("multiStatus")?.contains("Sync")) { multiStatusEvent("Sync OK.", true, true) }
+	}
+}
+
+private createChildDevices() {
+	logging("${device.displayName} - executing createChildDevices()","info")
+	addChildDevice(
+			"Fibaro Wall Plug USB",
+			"${device.deviceNetworkId}-2",
+			null,
+			[completedSetup: true, label: "${device.displayName} (CH2)", isComponent: false, componentName: "ch2", componentLabel: "Channel 2"]
+	)
+}
+
+private physicalgraph.app.ChildDeviceWrapper getChild(Integer childNum) {

```

Comment:
Remove the return type from this signature. This failed to compile in a previous DTH.